modules/standard/DynamicIters.chpl

```diff
@@ -154,6 +154,93 @@ where tag == iterKind.follower
   }
 }
 
+//************************* Multidimensional Dynamic iterator
+/*
+
+  :arg c: The domain to iterate over. The rank of the domain must be greater 
+          than one. If using a 1d domain, the user should use the existing
+          dynamic iterators (use DynamicIters). 
+  :type c: `domain`
+
+  :arg chunkSize: The size of chunks to be yielded to each thread. Must be
+                  greater than zero
+  :type chunkSize: `int`
+
+  :arg numTasks: The number of tasks to use. Must be >= zero. If this argument
+                 has the value 0, it will use the value indicated by
+                 ``dataParTasksPerLocale``.
+  :type numTasks: `int`
+
+  :arg parDim: The index of the dimension to parallelize across. Must be > 0.
+                Must be <= the rank of the domain ``c``. Defaults to 1. 
+  :type parDim: `int`
+
+  :yields: Indices of the domain ``c``
+
+  Given an input domain ``c``, each task is assigned slices of ``c``. The
+  chunks each have ``chunkSize`` slices in them (or the remaining iterations
+  if there are fewer than ``chunkSize``). This continues until there are no 
+  remaining iterations in the dimension of ``c`` indicated by ``parDim``. 
+
+  LIMITATIONS:
+  needs more testing with strided ranges
+  May or may not properly lead/follow with other domain iterators
+*/
+
+//This is the serial version of this iterator
+iter dynamic(c:domain, chunkSize:int, numTasks:int=0, parDim:int=1) 
+{
+  if debugDynamicIters then
+    writeln("Serial Multidimensional Dynamic Iterator, working with domain: ", c);
+  
+
+  for yieldTuple in c do yield yieldTuple;
+}
+
+
+//this is the leader
+iter dynamic(param tag:iterKind, c:domain, chunkSize:int, numTasks:int=0, parDim : int = 1) 
```

Comment:
I should have caught this earlier, but I think we want a "no doc" pragma on the leader/follower iterators, like the range versions.